src/datafusion/parser.rs

```diff
@@ -51,6 +49,15 @@ fn parse_file_type(s: &str) -> Result<String, ParserError> {
     Ok(s.to_uppercase())
 }
 
+fn ensure_not_set<T>(field: &Option<T>, name: &str) -> Result<(), ParserError> {
+    if field.is_some() {
+        return Err(ParserError::ParserError(format!(
+            "{name} specified more than once",
+        )));
+    }
+    Ok(())
+}
+
 // XXX SEAFOWL: removed the struct definitions here because we want to use
```

Comment:
I forgot we keep having to pull DF's parser changes and rebase ours on top of them, does DF still not have hooks to extend the parser?

Reply:
Yes I meant to look into that, but didn't.

Looks like it's easier now, but still some methods are just private so we'd still need to copy most of the stuff.